src/zkevm_specs/evm/execution/create.py

```diff
@@ -0,0 +1,161 @@
+from zkevm_specs.evm.opcode import Opcode
+from zkevm_specs.util.arithmetic import FQ, RLC
+from zkevm_specs.util.hash import EMPTY_CODE_HASH, EMPTY_HASH
+from zkevm_specs.util.param import (
+    GAS_COST_CREATE,
+    N_BYTES_GAS,
+    N_BYTES_U64,
+)
+from ...util import (
+    CALL_CREATE_DEPTH,
+)
+from ..instruction import Instruction, Transition
+from ..table import RW, CallContextFieldTag, AccountFieldTag, CopyDataTypeTag
+
+
+def create(instruction: Instruction):
+    # check opcode is CREATE or CREATE2
+    opcode = instruction.opcode_lookup(True)
+    is_create, is_create2 = instruction.pair_select(opcode, Opcode.CREATE, Opcode.CREATE2)
+    instruction.responsible_opcode_lookup(opcode)
+
+    # set the caller_id to the current rw_counter
+    callee_call_id = instruction.curr.rw_counter
+
+    # Stack parameters and result
+    value = instruction.stack_pop()
+    offset = instruction.stack_pop()
+    size = instruction.stack_pop()
+    return_contract_address = instruction.stack_push()
+
+    depth = instruction.call_context_lookup(CallContextFieldTag.Depth)
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    caller_address = instruction.call_context_lookup(CallContextFieldTag.CallerAddress)
+    nonce, nonce_prev = instruction.account_write(caller_address, AccountFieldTag.Nonce)
+    is_success = instruction.call_context_lookup(CallContextFieldTag.IsSuccess)
+    is_static = instruction.call_context_lookup(CallContextFieldTag.IsStatic)
+    reversion_info = instruction.reversion_info()
+    contract_address = instruction.generate_contract_address(caller_address, nonce)
+    return_contract_address = is_success * contract_address
```

Comment:
To make it look like a constraint there, I think we could try:

```python3
instruction.constrain_equal(return_contract_address, is_success * contract_address)
```

instead of reassign the variable.

Reply:
fixed in [e8d9f05](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/e8d9f05bc2a014be74442e84b907966de3af68e1)